src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect, useState } from 'react';
+import {
+  useDispatch, connect, ConnectedProps,
+} from 'react-redux';
+import { getFilteringTodosList } from '../../api/helper';
+import { RooTStateT, TODOSTYPE } from '../../api/interface';
+import { selectedUser } from '../../store/todosList';
+import { NavMenuTodo } from '../NavMenuTodo/NavMenuTodo';
+import { RemoveTodo } from '../RemoveTodo';
+import './TodoList.scss';
+
+const mapStateToProps = (state: RooTStateT) => ({
+  todos: getFilteringTodosList(state.todosState.todos, state.navMenuState),
+});
+
+const connector = connect(
+  mapStateToProps,
+);
+
+type Props = ConnectedProps<typeof connector>;
+
+const TodoList: React.FC<Props> = ({
+  todos,
+}) => {
+  const [updateList, setUpdateList] = useState<TODOSTYPE[]>(todos);
+  const [todoId, setTodoId] = useState<number>(0);
+
+  useEffect(() => {
+    setUpdateList(todos);
+  }, [todos.length]);
+
+  const dispatch = useDispatch();
+
+  const handleClickUser = (userId: number) => {
+    dispatch(selectedUser(userId));
+  };
+
+  return (
+    <>
+      <NavMenuTodo
+        todosList={updateList}
+        setTodosList={setUpdateList}
+      />
+      <div className="TodoList">
+        <h2>Todos:</h2>
+
+        <div className="TodoList__list-container">
+          <ul className="TodoList__list">
+            {updateList.map((todo: TODOSTYPE) => (
```

Comment:
```suggestion
            {updateList.map((todo) => (
```
У тебя прописан тип для всего списка, соответственно у каждого элемента тоже должен быть тип